storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -124,6 +126,7 @@
     protected int idToTaskBase;
     protected String hostname;
     private static final double msDurationFactor = 1.0 / TimeUnit.MILLISECONDS.toNanos(1);
+    private AtomicBoolean needToRefreshCreds = new AtomicBoolean(true);
```

Comment:
The initial value should be `false` to avoid unnecessary update at the beginning.

Reply:
Fixed